x-pack/test/security_solution_endpoint_api_int/apis/resolver/events.ts

```diff
@@ -41,107 +44,231 @@ export default function ({ getService }: FtrProviderContext) {
+            .post(`/api/endpoint/resolver/${tree.origin.id}/events?events=2`)
+            .set('kbn-xsrf', 'xxx')
+            .expect(200);
+          expect(body.events.length).to.eql(2);
+          compareArrays(tree.origin.relatedEvents, body.events);
+          expect(body.nextEvent).not.to.eql(null);
+
+          ({ body } = await supertest
+            .post(
+              `/api/endpoint/resolver/${tree.origin.id}/events?events=2&afterEvent=${body.nextEvent}`
+            )
+            .set('kbn-xsrf', 'xxx')
+            .expect(200));
+          expect(body.events.length).to.eql(2);
+          compareArrays(tree.origin.relatedEvents, body.events);
+          expect(body.nextEvent).to.not.eql(null);
+
+          ({ body } = await supertest
+            .post(
+              `/api/endpoint/resolver/${tree.origin.id}/events?events=2&afterEvent=${body.nextEvent}`
+            )
+            .set('kbn-xsrf', 'xxx')
+            .expect(200));
+          expect(body.events).to.be.empty();
+          expect(body.nextEvent).to.eql(null);
+        });
+
+        it('should return the first page of information when the cursor is invalid', async () => {
+          const { body }: { body: SafeResolverRelatedEvents } = await supertest
+            .post(`/api/endpoint/resolver/${tree.origin.id}/events?afterEvent=blah`)
+            .set('kbn-xsrf', 'xxx')
+            .expect(200);
+          expect(body.events.length).to.eql(4);
+          compareArrays(tree.origin.relatedEvents, body.events, true);
+          expect(body.nextEvent).to.eql(null);
+        });
+
+        it('should sort the events in descending order', async () => {
+          const { body }: { body: SafeResolverRelatedEvents } = await supertest
+            .post(`/api/endpoint/resolver/${tree.origin.id}/events`)
+            .set('kbn-xsrf', 'xxx')
+            .expect(200);
+          expect(body.events.length).to.eql(4);
+          // these events are created in the order they are defined in the array so the newest one is
+          // the last element in the array so let's reverse it
+          const relatedEvents = tree.origin.relatedEvents.reverse();
+          for (let i = 0; i < body.events.length; i++) {
+            expect(body.events[i].event?.category).to.equal(relatedEvents[i].event?.category);
+            expect(eventIDSafeVersion(body.events[i])).to.equal(relatedEvents[i].event?.id);
+          }
+        });
       });
+    });
 
-      it('should return the first page of information when the cursor is invalid', async () => {
-        const { body }: { body: SafeResolverRelatedEvents } = await supertest
-          .post(
-            `/api/endpoint/resolver/${entityID}/events?legacyEndpointID=${endpointID}&afterEvent=blah`
-          )
-          .set('kbn-xsrf', 'xxx')
-          .expect(200);
-        expect(body.entityID).to.eql(entityID);
-        expect(body.nextEvent).to.eql(null);
+    describe('kql events route', () => {
```

Comment:
These are all new. Once the frontend is switched over to use this route the above tests will be removed.